lms/views/api/blackboard/sync.py

```diff
@@ -1,17 +1,72 @@
 from pyramid.view import view_config
 
+from lms.models import Grouping
 from lms.security import Permissions
 
 
 class Sync:
     def __init__(self, request):
         self._request = request
+        self._assignment_service = self._request.find_service(name="assignment")
+        self._grouping_service = self._request.find_service(name="grouping")
+        self._course_service = self._request.find_service(name="course")
+        self._blackboard_api = self._request.find_service(name="blackboard_api_client")
+
+        self._tool_consumer_instance_guid = self._request.json["lms"][
+            "tool_consumer_instance_guid"
+        ]
```

Comment:
I think this'll crash if you call the API with no `lms.tool_consumer_instance_guid` param (and I noticed that there's no guard schema on the view either). It's not super important because this API is only ever called by our own frontend code (and only _can_ be called by our own frontend code, due to the necessary permissions). But can we add an issue or note to the Blackboard Groups project board to add validation to this endpoint?

If someone does make an invalid and/or unauthenticated request the code would not get as far as executing the code in the view method since the permission or view guard schema (if it had one) would fail. But I'm not sure whether the code in the view class's `__init__()` method would actually still get executed and crash? (I'm guessing not, but I'm not sure off the top of my head.)

(If there _is_ a crash due to this code being in the `__init__()` then you can move it into a lazy `@property` instead, so you can still have the deduplication without the potentially crashing `__init__()`.)